# Exercises for insertions and deletions

## Predict the effects of 10 short insertions and deletions you have collected. If the databases did not contain so many, invent the variations yourself

*As I encountered issues making reference for annovar, I would use the hg19 build from on*

1. prepare genome reference for annovar:
    - `./annotate_variation.pl -downdb -buildver hg19 refGene humandb/`
    - `./annotate_variation.pl --buildver hg19 --downdb seq humandb/hg19_seq`
    - `./retrieve_seq_from_fasta.pl humandb/hg19_refGene.txt -seqdir humandb/hg19_seq -format refGene -outfile humandb/hg19_refGeneMrna.fa`
2. As shown in exercise 5, there's one short insertions and one deletions available. I generated another 9 insertions and deletions in vcf format myself. The module to yield coding level codon- insertions or deletions without frameshif is in [github](https://github.com/brainfo/indelsCreater)
3. use convert2annovar.pl to convert the vcf to annovar input:  `./convert2annovar.pl -format vcf4 /mnt/data/hong/courses/varianceAnalysis/data/indel.vcf > /mnt/data/hong/courses/varianceAnalysis/data/indel.vcf.avinput`
4. use coding_change.pl to convert vcf to input required by the MutPred2.
5. Run MutPredIndel.
    - install MATLAB runtime: `./install -mode silent -agreeToLicense yes -destinationFolder /crex/proj/snic2022-6-176/nobackup/private/MATLAB/MATLAB_Runtime`
    - issue when running
    - alternative to web service: 
Output: MutPredIndel_result.txt

## Search for insertions, deletions or indels that cause RNA frameshift and have protein effects. If not available for your protein, invent variants yourself. Run predictions

## Compare the prediction to information from other sources, if available. Compare predictions from different tools.

Which sources? They are invented.
Another tool mentioned in the course: 
