# Exercises for Systematics

## HGNC and VGNC

### Find alias names and symbols.

*NLM1* is a gene symbol whoes gene model is [At4g19030.1](https://www.arabidopsis.org/servlets/TairObject?type=gene&name=At4g19030.1). It is annotated in the Arabidopsis thaliana (Mouse-ear cress) and encoding [A0A1P8B5K6 Â· A0A1P8B5K6_ARATH](https://www.uniprot.org/uniprotkb/A0A1P8B5K6/entry) (NOD26-like major intrinsic protein 1).

Therefore, I would analyze *ADA* in the following assignments.

alias names and symbols for gene *ADA*:

- [*ADA1*][1][2]
  
[1]: https://www.genenames.org/data/gene-symbol-report/#!/hgnc_id/HGNC:186
[2]: https://www.ncbi.nlm.nih.gov/gene/100

### Is there a mouse homologue?

Yes: [*Ada*](https://informatics.jax.org/marker/MGI:87916)

## MANE

### Find MANE transcript for your gene.

Search term on NIH nucleotide: `(ADA[Gene Name] AND srcdb_refseq[PROP] AND MANE Select[Keyword] AND Homo sapiens[Organism])` and on ensembl.
Or just copy from HGNC entry:

- NM_000022.4
- ENST00000372874.9

## RefSeq

### How many transcripts are there and how do they differ?

There are four transcript variants. They are different in their sequence span on a genome assembly.

## UniProt

### How many variants are there in SwissProt and of which types?

Download variants report json file from UniProt and count id and consequenceType:
| Total:           | 595 |
| :--------------- | :-- |
| missense         | 526 |
| frameshift       | 33  |
| stop gained      | 28  |
| -                | 5   |
| stop lost        | 2   |
| inframe deletion | 1   |

> for the id entry, there seems to be different system. Could this be redundant? i.e., COSMO id and rs id actually are the same allel. Therefore, which entry should be taken to count the variants?
> what is the "-" in the consequenceType entry?

## EC

### What is the EC number for the protein and what does it mean?

[EC 3.5.4.4: adenosine deaminase](https://enzyme.expasy.org/EC/3.5.4.4)

## HGVS

### How are frameshift variations systematically named?

Frameshift
