<!DOCTYPE html>
<html>
<head>
<title>5. Exercises for LOVD.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="lovd-assignment">LOVD assignment</h1>
<h2 id="analyze-from-lovd">Analyze from LOVD</h2>
<ol>
<li>How many genes are totally in this LOVD installation?
In total: 4,641,475 variants in 1,801,643 individuals in 20 LOVD installations.
In the main database Global Variome shared LOVD, there are 23034 genes.</li>
</ol>
<h2 id="analyze-lovd-for-your-gene">Analyze LOVD for your gene</h2>
<ol>
<li>
<p>If there are several databases for your gene, how many cases do they contain?</p>
<ul>
<li>ClinVar at NCBI: 558</li>
<li>Global Variome shared LOVD: 59</li>
</ul>
</li>
<li>
<p>How many (public) variants are there (total, unique)?
Fron Global Variome shared LOVD:</p>
<ul>
<li>Total: 97</li>
<li>Unique: 59</li>
</ul>
</li>
<li>
<p>What are the reference sequences? Do they use MANE?
The variants shown are described using the NM_000022.2 transcript reference sequence.
No, the reference sequence is not MANE.</p>
</li>
<li>
<p>Which is the 'main' disease connected to the gene (name, acronym, OMIM ID)?
If the 'main' is the diseases with OMIM ID and more than one phenotypic records with most Individuals, then immunodeficiency, severe combined, due to ADA deficiency is the main disease connected to ADA.</p>
<ul>
<li>name:immunodeficiency, severe combined, due to ADA deficiency</li>
<li>acronym: SCIDADA</li>
<li>OMIM ID: 102700</li>
</ul>
</li>
<li>
<p>How many individuals and phenotypes with this disease are in the database?
31 individuals and 1 phenotype entries for this disease</p>
</li>
</ol>
<h2 id="select-an-amino-acid-substitution">Select an amino acid substitution</h2>
<p>I select p.(His15Asp):</p>
<ol>
<li>What are the changes in RNA and DNA, are these observed or predicted?
<ul>
<li>DNA change: c.43C&gt;G: observed</li>
<li>RNA change:  r.(?): RNA has not been analysed but an effect other then that caused by the change at DNA level is not expected</li>
</ul>
</li>
<li>What is the gender of the individual?
No record fo the gender of the individual.</li>
<li>Which disease was this individual diagnosed with (see phenotype)?
No record fo the phenotypes.</li>
<li>Complement the variants you collected in database exercise, so that you have ten different amino acid substitutions, and ten short insertions and deletions (if enough cases available).
The variants collected:
<ul>
<li>amino acid substitutions:
<ol>
<li>Variant #0000434892 (NC_000020.10:g.43264920G&gt;C, ADA(NM_000022.2):c.43C&gt;G)</li>
<li>Variant #0000434893 (NC_000020.10:g.43264905C&gt;T, ADA(NM_000022.2):c.58G&gt;A)</li>
<li>Variant #0000434894 (NC_000020.10:g.43255239C&gt;A, ADA(NM_000022.2):c.220G&gt;T)</li>
<li>Variant #0000434895 (NC_000020.10:g.43255233G&gt;A, ADA(NM_000022.2):c.226C&gt;T)</li>
<li>Variant #0000434896 (NC_000020.10:g.43255211G&gt;T, ADA(NM_000022.2):c.248C&gt;A)</li>
<li>Variant #0000434897 (NC_000020.10:g.43255158G&gt;A, ADA(NM_000022.2):c.301C&gt;T)</li>
<li>Variant #0000434898 (NC_000020.10:g.43255157C&gt;T, ADA(NM_000022.2):c.302G&gt;A)</li>
<li>Variant #0000434899 (NC_000020.10:g.43255157C&gt;A, ADA(NM_000022.2):c.302G&gt;T)</li>
<li>Variant #0000434900 (NC_000020.10:g.43255139A&gt;G, ADA(NM_000022.2):c.320T&gt;C)</li>
<li>Variant #0000434901 (NC_000020.10:g.43254303C&gt;T, ADA(NM_000022.2):c.385G&gt;A)</li>
</ol>
</li>
<li>short insertions and deletions
In the database, there's only one deletions, and it's not connected any disease. No insertion is recorded.<br>
The deletion recorded is Variant #0000000894 (NC_000020.10:g.43255242T&gt;C, ADA(NM_000022.2):c.219-2A&gt;G)
I would say if not confined with the diseas report, I'd like to take the records from SwissProt, see the output file ADA_variants_rs.csv</li>
</ul>
</li>
</ol>

</body>
</html>
