<!DOCTYPE html>
<html>
<head>
<title>9. Exercises for pathogenicity.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="exercises-for-pathogenicity-prediction">Exercises for pathogenicity prediction</h1>
<ol>
<li>
<p>Use the variants you have collected in previous exercises. Predict the tolerance of variants with FATHMM and AlphaMissense. If the databases did not contain at least 10 amino acid substitutions, just make up variants (the original amino acid has to be correct).</p>
<p>Input rs id from uniprot, the preprocessing of the uniprot json were done in the Exercises 3 and in the uniprot_json.ipynb
then,</p>
<pre class="hljs"><code><div>cut -d&quot;,&quot; -f1 ADA_variant_rs.csv | sed '1d' &gt; ADA_variant_rsid.txt
</div></code></pre>
<ul>
<li>FATHMM results are from web tool
<blockquote>
<p>The interpretation from the output. I suppose the HMM weights D. (float.) is the  relative frequencies of disease-associated but what is HMM weights O.?</p>
</blockquote>
</li>
<li>AlphaMissense are from VEP AlphaMissense plugin in Docker</li>
</ul>
<pre class="hljs"><code><div>docker run -t -i -u root -v /mnt/data/hong/courses/varianceAnalysis/data:/data ensemblorg/ensembl-vep
INSTALL.pl -a cf -s homo_sapiens -y GRCh38
tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz
vep -i ADA_variant_rsid.txt --plugin AlphaMissense,file=AlphaMissense_hg38.tsv.gz -o ADA_variant_rsid_vep_aMissense.txt --cache --force_overwrite
</div></code></pre>
</li>
<li>
<p>Correlate the predictions to clinical data and allele frequencies, when available. Indicate sources of data. Compare the two methods.</p>
<p>Again, the observed clinical data and allele frequencies are fields (clinicalSignificances, freqiencies with population name MAF) in the features in the uniprot json file. As basically I worked with the variants there, I have the corresponding data.</p>
<ol>
<li>
<p>FATHMM</p>
<ul>
<li>The correlation with clinical Significances could be test with Chisquare Test on contingency table between clinicalSignificances and Prediction:
The contingency table:
| Prediction                        | DAMAGING | NA    |
| :-------------------------------- | -------- | ----- |
| clinicalSignificances             |          |       |
| Likely benign                     | 370      | 569   |
| Likely pathogenic                 | 0        | 155   |
| NA                                | 821      | 13079 |
| Pathogenic                        | 1451     | 368   |
| Variant of uncertain significance | 311      | 2596  |
<em>here we remove the one with NA clinicalSignificances records</em>
| statistic | 2393.111833770712 |
| p-------e | 0---------------0 |</li>
<li>the correlation with the MAF could be inferred using pearson correlation (linear regression) between MAF and HMM Prob. M..
| rvalue | 0.6230381134654781      |
| :----- | ----------------------- |
| pvalue | 2.7072513044468046e-202 |</li>
</ul>
</li>
<li>
<p>AlphaMissense</p>
<ul>
<li>The correlation with clinical Significances could be test with Chisquare Test on contingency table between clinicalSignificances and am_class.
The contingency table:
| IMPACT                            | HIGH | LOW | MODERATE | MODIFIER |
| :-------------------------------- | :--- | :-- | :------- | :------- |
| clinicalSignificances             |      |     |          |          |
| Likely benign                     | 0    | 136 | 266      | 537      |
| Likely pathogenic                 | 0    | 0   | 97       | 58       |
| NA                                | 314  | 518 | 5831     | 7237     |
| Pathogenic                        | 35   | 67  | 799      | 918      |
| Variant of uncertain significance | 0    | 57  | 1256     | 1594     |
<em>here we remove the one with NA clinicalSignificances records</em>
| statistic | 412.55607384818313 |
| pvaluee | 2.8636735807461966e-83 |</li>
</ul>
</li>
<li>
<p>Results interpretation:</p>
<ul>
<li>AlphaMissense gave more stratified impact. It reports most variants as modifier. If we take out the variants that are not recorded in clinvar, the all the predicted high are pathogenic. And most of the predicted low are likely benign. These might drive the contingency table test to be significant.</li>
<li>FATHMM predicted DAMAGING is low in information.</li>
<li>FATHMM HMM Prob. M. is correlated to the MAF</li>
<li>So far I didn't come up with the comparible observed score to the AlphaMissense predicted pathogenicity score</li>
</ul>
</li>
</ol>
</li>
</ol>

</body>
</html>
