# Exercises for pathogenicity prediction

1. Use the variants you have collected in previous exercises. Predict the tolerance of variants with FATHMM and AlphaMissense. If the databases did not contain at least 10 amino acid substitutions, just make up variants (the original amino acid has to be correct).

    Input rs id from uniprot, the preprocessing of the uniprot json were done in the Exercises 3 and in the uniprot_json.ipynb
    then,

    ```{sh}
    cut -d"," -f1 ADA_variant_rs.csv | sed '1d' > ADA_variant_rsid.txt
    ```

    - FATHMM results are from web tool
        > The interpretation from the output. I suppose the HMM weights D. (float.) is the  relative frequencies of disease-associated but what is HMM weights O.?  
    - AlphaMissense are from VEP AlphaMissense plugin in Docker

    ```{sh}
    docker run -t -i -u root -v /mnt/data/hong/courses/varianceAnalysis/data:/data ensemblorg/ensembl-vep
    INSTALL.pl -a cf -s homo_sapiens -y GRCh38
    tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz
    vep -i ADA_variant_rsid.txt --plugin AlphaMissense,file=AlphaMissense_hg38.tsv.gz -o ADA_variant_rsid_vep_aMissense.txt --cache --force_overwrite
    ```

2. Correlate the predictions to clinical data and allele frequencies, when available. Indicate sources of data. Compare the two methods.

    Again, the observed clinical data and allele frequencies are fields (clinicalSignificances, freqiencies with population name MAF) in the features in the uniprot json file. As basically I worked with the variants there, I have the corresponding data.  
    1. FATHMM
        - The correlation with clinical Significances could be test with Chisquare Test on contingency table between clinicalSignificances and Prediction:
        The contingency table:
        | Prediction                        | DAMAGING | NA    |
        | :-------------------------------- | -------- | ----- |
        | clinicalSignificances             |          |       |
        | Likely benign                     | 370      | 569   |
        | Likely pathogenic                 | 0        | 155   |
        | NA                                | 821      | 13079 |
        | Pathogenic                        | 1451     | 368   |
        | Variant of uncertain significance | 311      | 2596  |
        *here we remove the one with NA clinicalSignificances records*
        | statistic | 2393.111833770712 |
        | p-------e | 0---------------0 |
        - the correlation with the MAF could be inferred using pearson correlation (linear regression) between MAF and HMM Prob. M..
        | rvalue | 0.6230381134654781      |
        | :----- | ----------------------- |
        | pvalue | 2.7072513044468046e-202 |

    2. AlphaMissense
        - The correlation with clinical Significances could be test with Chisquare Test on contingency table between clinicalSignificances and am_class.
        The contingency table:
        | IMPACT                            | HIGH | LOW | MODERATE | MODIFIER |
        | :-------------------------------- | :--- | :-- | :------- | :------- |
        | clinicalSignificances             |      |     |          |          |
        | Likely benign                     | 0    | 136 | 266      | 537      |
        | Likely pathogenic                 | 0    | 0   | 97       | 58       |
        | NA                                | 314  | 518 | 5831     | 7237     |
        | Pathogenic                        | 35   | 67  | 799      | 918      |
        | Variant of uncertain significance | 0    | 57  | 1256     | 1594     |
        *here we remove the one with NA clinicalSignificances records*
        | statistic | 412.55607384818313 |
        | pvaluee | 2.8636735807461966e-83 |

    3. Results interpretation:  
        - AlphaMissense gave more stratified impact. It reports most variants as modifier. If we take out the variants that are not recorded in clinvar, the all the predicted high are pathogenic. And most of the predicted low are likely benign. These might drive the contingency table test to be significant.
        - FATHMM predicted DAMAGING is low in information.
        - FATHMM HMM Prob. M. is correlated to the MAF
        - So far I didn't come up with the comparible observed score to the AlphaMissense predicted pathogenicity score
